test-clients/src/main/java/com/hedera/services/bdd/suites/token/UniqueTokenManagementSpecs.java

```diff
@@ -550,6 +559,129 @@ private HapiApiSpec uniqueTokenMintReceiptCheck() {
+								)
+								.logged()
+				);
+	}
+
+	private HapiApiSpec validateTokenNftInfosOutOfRange() {
+		return defaultHapiSpec("ValidateTokenNftInfosOutOfRange")
+				.given(
+						newKeyNamed(SUPPLY_KEY),
+						cryptoCreate(TOKEN_TREASURY),
+						tokenCreate(NFT)
+								.initialSupply(0)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.supplyType(TokenSupplyType.INFINITE)
+								.supplyKey(SUPPLY_KEY)
+								.treasury(TOKEN_TREASURY)
+				).when(
+						mintToken(NFT, List.of(
+								ByteString.copyFromUtf8("some metadata"),
+								ByteString.copyFromUtf8("some metadata2"),
+								ByteString.copyFromUtf8("some metadata3")
+						))
+				).then(
+						getTokenNftInfos(NFT, 0, 6)
+								.hasCostAnswerPrecheck(QUERY_RANGE_LIMIT_EXCEEDED)
+				);
+	}
+
+	private HapiApiSpec failsWithTokenWithoutNfts() {
+		return defaultHapiSpec("FailsWithTokenWithoutNfts")
+				.given(
+						newKeyNamed(SUPPLY_KEY),
+						cryptoCreate(TOKEN_TREASURY),
+						tokenCreate(NFT)
+								.initialSupply(0)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.supplyType(TokenSupplyType.INFINITE)
+								.supplyKey(SUPPLY_KEY)
+								.treasury(TOKEN_TREASURY)
+				).when()
+				.then(
+						getTokenNftInfos(NFT, 0, 2)
+								.hasCostAnswerPrecheck(QUERY_RANGE_LIMIT_EXCEEDED)
+				);
+	}
+
+	private HapiApiSpec failsWithTokenNftsInvalidQueryBoundaries() {
+		return defaultHapiSpec("FailsWithTokenNftsInvalidQueryBoundaries")
+				.given(
+						newKeyNamed(SUPPLY_KEY),
+						cryptoCreate(TOKEN_TREASURY),
+						tokenCreate(NFT)
+								.initialSupply(0)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.supplyType(TokenSupplyType.INFINITE)
+								.supplyKey(SUPPLY_KEY)
+								.treasury(TOKEN_TREASURY)
+				).when().then(
+						getTokenNftInfos(NFT, 2, 0)
+								.hasCostAnswerPrecheck(QUERY_RANGE_LIMIT_EXCEEDED),
+						getTokenNftInfos(NFT, 0, 100000000)
+								.hasCostAnswerPrecheck(QUERY_RANGE_LIMIT_EXCEEDED)
+				);
+	}
+
+	private HapiApiSpec failsWithDeletedTokenNft() {
+		return defaultHapiSpec("FailsWithDeletedTokenNft")
+				.given(
+						newKeyNamed(SUPPLY_KEY),
+						newKeyNamed("nftAdmin"),
+						cryptoCreate(TOKEN_TREASURY),
+						tokenCreate(NFT)
+								.initialSupply(0)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.supplyType(TokenSupplyType.INFINITE)
+								.supplyKey(SUPPLY_KEY)
+								.adminKey("nftAdmin")
+								.treasury(TOKEN_TREASURY)
+				).when(
+						tokenDelete(NFT)
+				)
+				.then(
+						getTokenNftInfos(NFT, 0, 2)
+								.hasCostAnswerPrecheck(TOKEN_WAS_DELETED)
+				);
+	}
+
+	public HapiApiSpec failsWithFungibleTokenGetNftInfos() {
+		return defaultHapiSpec("FailsWithFungibleTokenGetNftInfos")
+				.given(
+						cryptoCreate(TOKEN_TREASURY).balance(0L)
+				).when(
+						tokenCreate(A_TOKEN)
+								.initialSupply(1_000)
+								.treasury(TOKEN_TREASURY)
+				).then(
+						getTokenNftInfos(A_TOKEN, 0, 2)
```

Comment:
The response code for this query is not appropriate

Reply:
I've traversed all the available responses and to my mind NOT_SUPPORTED is the best choice here